File: CallX2/callx_gif_work/feature-chat/src/main/java/com/callx/app/contacts/ContactShareAdapter.java
package com.callx.app.contacts;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import com.callx.app.chat.R;
import com.callx.app.models.User;

import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;

public class ContactShareAdapter
        extends RecyclerView.Adapter<ContactShareAdapter.ContactVH> {

    public interface OnContactShareListener {
        void onShareToContact(User contact);
    }

    private final List<User>              contacts;
    private final OnContactShareListener  listener;

    public ContactShareAdapter(List<User> contacts, OnContactShareListener listener) {
        this.contacts = contacts;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ContactVH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
            .inflate(R.layout.item_contact_share, parent, false);
        return new ContactVH(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ContactVH h, int pos) {
        User contact = contacts.get(pos);
        h.tvName.setText(contact.name != null ? contact.name : "User");
        String avatarUrl = (contact.thumbUrl != null && !contact.thumbUrl.isEmpty())
            ? contact.thumbUrl : contact.photoUrl;
        if (avatarUrl != null && !avatarUrl.isEmpty()) {
            Glide.with(h.itemView.getContext())
                .load(avatarUrl)
                .apply(RequestOptions.circleCropTransform())
                .placeholder(R.drawable.ic_person)
                .into(h.ivAvatar);
        } else {
            h.ivAvatar.setImageResource(R.drawable.ic_person);
        }

        h.itemView.setOnClickListener(v -> listener.onShareToContact(contact));
    }

    @Override public int getItemCount() { return contacts.size(); }

    static class ContactVH extends RecyclerView.ViewHolder {
        CircleImageView ivAvatar;
        TextView        tvName;

        ContactVH(View v) {
            super(v);
            ivAvatar = v.findViewById(R.id.iv_share_contact_avatar);
            tvName   = v.findViewById(R.id.tv_share_contact_name);
        }
    }
}

File: CallX2/feature-chat/src/main/java/com/callx/app/adapters/ContactShareAdapter.kt
package com.callx.app.adapters

import android.view.*
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.callx.app.chat.R
import com.callx.app.models.User
import de.hdodenhof.circleimageview.CircleImageView

class ContactShareAdapter(
    private val contacts: List<User>,
    private val listener: OnContactShareListener
) : RecyclerView.Adapter<ContactShareAdapter.ContactVH>() {

    interface OnContactShareListener {
        fun onShareToContact(contact: User)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContactVH {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item_contact_share, parent, false)
        return ContactVH(v)
    }

    override fun onBindViewHolder(h: ContactVH, pos: Int) {
        val c = contacts[pos]
        h.tvName.text = c.name ?: "User"
        val avatarUrl = c.thumbUrl?.takeIf { it.isNotEmpty() } ?: c.photoUrl
        if (!avatarUrl.isNullOrEmpty()) {
            Glide.with(h.itemView.context).load(avatarUrl)
                .apply(RequestOptions.circleCropTransform())
                .placeholder(R.drawable.ic_person).into(h.ivAvatar)
        } else h.ivAvatar.setImageResource(R.drawable.ic_person)
        h.itemView.setOnClickListener { listener.onShareToContact(c) }
    }

    override fun getItemCount() = contacts.size

    class ContactVH(v: View) : RecyclerView.ViewHolder(v) {
        val tvName:   TextView        = v.findViewById(R.id.tv_name)
        val ivAvatar: CircleImageView = v.findViewById(R.id.iv_avatar)
    }
}
